src/main/java/com/linkedin/replica/wall/database/handlers/impl/ArangoWallHandler.java

```diff
@@ -296,36 +293,25 @@ public boolean addComment(Comment comment) throws ArangoDBException{
      * @param args
      * @return
      */
-    public boolean editComment(HashMap<String, Object> args) throws ArangoDBException {
-        boolean response = false;
-        boolean isString = false;
-        Class commentClass = Comment.class;
-        Field[] commentFields = commentClass.getDeclaredFields();
 
-            String query = "FOR c IN " + commentsCollection + " FILTER c._key == @key UPDATE c with {";
-            for (String key : args.keySet()) {
-                if(!key.equals("commentId")){
-                    query += key + ":";
-                    for (int i = 0; i<commentFields.length; i++) {
-                        if (key.equals(commentFields[i].getName()) && String.class.isAssignableFrom(commentFields[i].getType())){
-                            query += "\"" + args.get(key) + "\" ";
-                            isString = true;
-                            break;
-                        }
-                    }
-                    if(!isString){
-                        query += args.get(key);
-                        isString = false;
-                    }
-                    query+=",";
+    public boolean editComment(HashMap<String, Object> args) throws ArangoDBException {
+        Map<String, Object> bindVars = new HashMap<>();
+         int counter = 0;
+         boolean response = false;
+         String query = "FOR c IN " + commentsCollection + " FILTER c._key == @key UPDATE c with {";
+         bindVars.put("key",args.get("commentId").toString());
+         for (String key : args.keySet()) {
+             if(!key.equals("commentId")){

```

Comment:
Add not equal to authorId as well

Reply:
If I added authorId in the conditon, the author Id will not be added to the database, the model Id should not be added because It is auto generated and not added as a parameter in the database.